extensions/hibernate-search-elasticsearch/deployment/src/main/java/io/quarkus/hibernate/search/elasticsearch/HibernateSearchElasticsearchProcessor.java

```diff
@@ -97,37 +97,34 @@ void setRuntimeConfig(HibernateSearchElasticsearchRecorder recorder,
     }
 
     private static void checkConfig(HibernateSearchElasticsearchBuildTimeConfig buildTimeConfig) {
-        if (buildTimeConfig.additionalBackends.defaultBackend.isPresent()) {
-            String defaultBackend = buildTimeConfig.additionalBackends.defaultBackend.get();
-            // we have a default named backend
-            if (buildTimeConfig.defaultBackend.version.isPresent()) {
-                throw new ConfigurationError(
-                        "quarkus.hibernate-search.elasticsearch.default-backend cannot be used in conjunction with a default backend configuration.");
-            }
-            if (!buildTimeConfig.additionalBackends.backends.containsKey(defaultBackend)) {
-                throw new ConfigurationError(
-                        "The default backend defined does not exist: " + defaultBackend);
-            }
-        } else {
-            // we are in the default backend case
+        if (buildTimeConfig.namedBackends.backends.isEmpty()) {
+            // There isn't any named backend: the default backend will be used.
+
+            // we validate that the version is present
             if (!buildTimeConfig.defaultBackend.version.isPresent()) {
                 throw new ConfigurationError(
                         "The Elasticsearch version needs to be defined via the quarkus.hibernate-search.elasticsearch.version property.");
             }
-        }
+        } else {
+            // There is at least one named backend: we disallow using the default backend.
+            // Theoretically we could allow using it,
+            // but then we would be unable to detect whether the default backend is used
+            // (and thus absolutely needs the version to be configured)
```

Comment:
Not sure I follow you here. We make the version mandatory for named backend so maybe we should just make it mandatory for everything, default backend included? And then we could keep the named backend active?

Reply:
If we do that, then people using named backends exclusively (which, IMO, is what they should do) will have to set the version for the default backend even if they don't use it. I can do that if you want, but I'm not sure it's a better solution.

Reply:
I fixed this according to our discussion on Zulip, rebased and force-pushed. Let's wait for CI, but then as far as I know it can be merged.